### PR TITLE
Refactor pyi_generator to use AST directly

### PR DESCRIPTION
**Pull request by masenf**
* Dynamically import modules to `eval` type hints
* Use a `NodeTransformer` to walk the abstract syntax tree of the original module
* Using AST data and dynamic runtime data, construct a new `.create` method for `Component` classes that includes all props declared as pydantic fields.
* Remove annotated assignment targets
* Remove assignments entirely (unless it's assigning `typing.Any` as an ImportError fallback)
* Add docstrings and type annotations throughout the script

### CODE REVIEW

**Comment by masenf** [comment]
Docstrings for Accordion.create and ColorModeButton.create are getting generated weird because some of the props in `create` overlap with props defined in the class.

File: scripts/pyi_generator.py [hunk 1]
```diff
@@ -177,61 +46,457 @@
...
+        defaults=[],
+    )
+    definition = ast.FunctionDef(
+        name="create",
+        args=create_args,
+        body=[
+            ast.Expr(
+                value=ast.Constant(value=_generate_docstrings(all_classes, all_props))
+            ),
+        ],
```
Comment (Lendemor) [inline scripts/pyi_generator.py:323]: Should probably include an `ast.Ellipsis` after the docstring for cleanliness ?

File: scripts/pyi_generator.py [hunk 2]
```diff
@@ -244,215 +509,57 @@ class PyiGenerator:
...
+                pyi_content.append(formatted_line)
 
-    def _scan_file(self, file):
-        self.current_module_path = os.path.splitext(file)[0]
-        module_import = os.path.splitext(os.path.join(self.root, file))[0].replace(
-            "/", "."
-        )
+        pyi_path = module_path.with_suffix(".pyi")
+        pyi_path.write_text("\n".join(pyi_content))
+        logger.info("Wrote %s", pyi_path)
```
Comment (Lendemor) [inline scripts/pyi_generator.py:534]: Any reason that prevent using f-string here?

**Review by Lendemor — CHANGES_REQUESTED** [review]
Some small additions/changes, but I love it, should make it much easier to extend later if needed.

File: scripts/pyi_generator.py (hunk 2 above)
Comment (masenf) [reply scripts/pyi_generator.py:534]: no, i can fix that

**Review by Lendemor — APPROVED** [review]

**Pull request merged**